### DISCUSSION TITLE
Question - Why does building git always package /cmd/git.exe from the sdk?

### DISCUSSION THREAD

**Discussion by uberDoward** [git-for-windows/git#5096] [Q&A]
I have to be missing something very basic here.

1. Install git-sdk
2. Open git-bask from the git-sdk
3. sdk init git
4. sdk init build-extra
5. cd /usr/src/git
6. Checkout any tag (2.45.2 in my case)
7. Make custom updates (any)
8. make all && make install && git --version (now shows my new version)
9. /usr/src/build-extra/installer/release.sh 2.45.2.custom
10. Inno package created, but contains the 2.46.0 version that came with the git-sdk, rather than my custom built git

How do I tell Inno to package up my built git, vs the git included in /cmd ?

**Comment by dscho** [comment]
> the git included in /cmd ?

This is not actually the actual `git.exe`. It's merely the [Git wrapper](https://github.com/git-for-windows/git/wiki/Git-wrapper), and if you are looking at its embedded file version, it will reflect the version of the `mingw-w64-x86_64-git` package, from which it originates. However, after running the installer, `git --version` should reflect the version you installed in the SDK via `make install`.

**Reply by uberDoward** [reply]
And the git-wrapper is included / versioned with the git-sdk?  Well, that explains the 44Kb vs 21Mb difference I was seeing between cmd/git.exe and the built git, lol.

Does git --version report the git-wrapper version, or the actual git executable that is used?

**Reply by dscho** [reply]
> And the git-wrapper is included / versioned with the git-sdk?

No, it is versioned with the `mingw-w64-x86_64-git` package, which is installed in the SDK.

> Does git --version report the git-wrapper version, or the actual git executable that is used?

Since `git --version`, when run via `/cmd/git.exe --version`, simply sets up a couple of environment variables and then executes `/mingw64/bin/git.exe` with the same command-line options (as explained [here](https://github.com/git-for-windows/git/wiki/Git-wrapper)), it would report the actual executable's version.

**Reply by uberDoward** [reply]
Thank you!  Let me go work on this more and see what the what is - that definitely helps clarify several points for me!
